### PR TITLE
Update device virtual's private config file

### LINKED ISSUE #11: Upgrade device virtual config files
The files have changed upstream, mostly as a result of moving common config to the config bootstrapper.

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR will resolve #11

## Testing
Manually run snap test with the edgex-config-provider snap built from this PR:
```
$ LOCAL_SNAP=~/Desktop/edgex671/edgex-config-provider/edgex-config-provider-example_2.3_amd64.snap go test -v ./test/suites/edgex-config-provider/
--- FAIL: TestConfigProvider (263.25s)
    --- FAIL: TestConfigProvider/app-service-configurable (25.27s)
    --- FAIL: TestConfigProvider/app-rfid-llrp-inventory (22.11s)
    --- FAIL: TestConfigProvider/device-gpio (25.56s)
    --- FAIL: TestConfigProvider/device-modbus (22.60s)
    --- FAIL: TestConfigProvider/device-mqtt (20.91s)
    --- FAIL: TestConfigProvider/device-rest (20.37s)
    --- FAIL: TestConfigProvider/device-rfid-llrp (21.04s)
    --- FAIL: TestConfigProvider/device-snmp (29.03s)
    --- FAIL: TestConfigProvider/device-usb-camera (33.70s)
    --- FAIL: TestConfigProvider/device-virtual (20.63s)
    --- FAIL: TestConfigProvider/device-onvif-camera (22.02s)
FAIL
```
Logs:
Many services fail with the error message 'unable to register service with consul' which is unrelated to this PR. Unfortunately, this error prevents the testing of this PR:
```
Mär 27 16:12:09 ubuntu edgex-device-virtual.device-virtual[93092]: level=WARN ts=2023-03-27T14:12:09.276703694Z app=device-virtual source=registry.go:118 msg="could not register service with Registry: unable to register service with consul: Service information not set"
Mär 27 16:08:58 ubuntu edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory[85574]: level=WARN ts=2023-03-27T14:08:58.509378966Z app=app-rfid-llrp-inventory source=registry.go:118 msg="could not register service with Registry: unable to register service with consul: Service information not set"
Mär 27 16:09:25  ubuntu edgex-device-gpio.device-gpio[86414]: level=WARN ts=2023-03-27T14:09:24.922008785Z app=device-gpio source=registry.go:118 msg="could not register service with Registry: unable to register service with consul: Service information not set"
...
```
ASC service fails due to `failed to read configuration file` error which is unrelated to this PR:
```
Mär 27 16:08:34 ubuntu edgex-app-service-configurable.app-service-configurable[84712]: level=ERROR ts=2023-03-27T14:08:34.372892606Z app=app-http-export source=bootstrap.go:48 msg="failed to read configuration file /var/snap/edgex-app-service-configurable/1816/config/res/http-export/configuration.yaml: open /var/snap/edgex-app-service-configurable/1816/config/res/http-export/configuration.yaml: no such file or directory"
```

### CODE REVIEW

**Comment by farshidtz** [comment]
The serialization format has changed to yaml recently: https://github.com/edgexfoundry/device-virtual-go/pull/368

**Review by farshidtz — CHANGES_REQUESTED** [review]
The serialization format has recently changed to yaml: https://github.com/edgexfoundry/device-virtual-go/pull/368

This also affects https://github.com/canonical/edgex-config-provider/blob/main/amend-startup-msg.sh

File: amend-startup-msg.sh [hunk 1]
```diff
@@ -5,6 +5,6 @@
 
 path=$1
 
-find $path -type f -name 'configuration.toml' | xargs \
+find $path -type f -name 'configuration.yaml' | xargs \
```
Comment (farshidtz) [inline amend-startup-msg.sh:8]: @MonicaisHer have you tested this? It injects toml syntax in a yaml file.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: amend-startup-msg.sh (hunk 1 above)
Comment (MonicaisHer) [reply amend-startup-msg.sh:8]: I forgot they have different syntax, so I didn't test it. Updated now.

**Review by farshidtz — APPROVED** [review]
Looks good. Thanks.
Need to merge these changes into main branch as well.

**Comment by farshidtz** [comment]
As a quick workaround to the registry issue, you can set [EDGEX_USE_REGISTRY](https://docs.edgexfoundry.org/3.0/microservices/configuration/CommonEnvironmentVariables/#edgex_use_registry) to false.

**Comment by MonicaisHer** [comment]
I tried out this PR using the ASC snap while setting EDGEX_USE_REGISTRY to false:
```
$ LOCAL_SNAP=~/Desktop/edgex671/edgex-config-provider/edgex-config-provider-example_2.3_amd64.snap go test -v ./test/suites/edgex-config-provider/
2023/04/03 12:03:50 [CLEAN]
2023/04/03 12:03:50 [exec] sudo snap remove --purge edgexfoundry
2023/04/03 12:03:50 [stderr] snap "edgexfoundry" is not installed
2023/04/03 12:03:50 [exec] sudo snap remove --purge edgex-config-provider-example
2023/04/03 12:03:50 [stderr] snap "edgex-config-provider-example" is not installed
2023/04/03 12:03:50 [SETUP]
2023/04/03 12:03:50 [exec] sudo snap install --dangerous /home/mengyi/Desktop/edgex671/edgex-config-provider/edgex-config-provider-example_2.3_amd64.snap
2023/04/03 12:03:52 [stdout] edgex-config-provider-example 2.3 installed
2023/04/03 12:03:52 [exec] sudo snap install edgexfoundry --channel=latest/edge
2023/04/03 12:04:25 [stdout] edgexfoundry (edge) 3.0.0-dev.117 from Canonical** installed
2023/04/03 12:04:25 Retry 1/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command), 8200 (vault), 8500 (consul), 6379 (redis)
2023/04/03 12:04:26 Retry 2/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command)
2023/04/03 12:04:27 Retry 3/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command)
2023/04/03 12:04:28 Retry 4/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command)
2023/04/03 12:04:29 Retry 5/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command)
2023/04/03 12:04:30 Retry 6/180: Waiting for ports: 59880 (core-data), 59882 (core-command)
2023/04/03 12:04:30 [exec] sudo snap set edgexfoundry config.edgex-use-registry='false'
=== RUN   TestConfigProvider
=== RUN   TestConfigProvider/app-service-configurable
    exec.go:19: [exec] sudo snap remove --purge edgex-app-service-configurable
    exec.go:101: [stderr] snap "edgex-app-service-configurable" is not installed
    exec.go:19: [exec] sudo snap install edgex-app-service-configurable --channel=latest/edge
    exec.go:101: [stdout] edgex-app-service-configurable (edge) 3.0.0-dev.29 from Canonical** installed
    exec.go:19: [exec] sudo snap set edgex-app-service-configurable config.edgex-use-registry='false'
2023/04/03 12:04:40 [exec] sudo snap set edgex-app-service-configurable profile='http-export'
    exec.go:19: [exec] sudo snap connect edgex-app-service-configurable:app-service-configurable-config edgex-config-provider-example:app-service-configurable-config
    exec.go:19: [exec] sudo snap start --enable edgex-app-service-configurable
    exec.go:101: [stdout] Started.
    config.go:190: Retry 1/10: Waiting for startup message: CONFIG BY EXAMPLE PROVIDER
    exec.go:19: [exec] sudo journalctl --since "2023-04-03 12:04:30" --no-pager | grep "edgex-app-service-configurable"|| true
    config.go:195: Found startup message: "CONFIG BY EXAMPLE PROVIDER"
    exec.go:19: [exec] (sudo journalctl --since "2023-04-03 12:04:30" --no-pager | grep "edgex-app-service-configurable"|| true) > edgex-app-service-configurable.log
Wrote snap logs to /home/mengyi/Desktop/edgex671/edgex-snap-testing/test/suites/edgex-config-provider/edgex-app-service-configurable.log
    exec.go:19: [exec] sudo snap remove --purge edgex-app-service-configurable
    exec.go:101: [stdout] edgex-app-service-configurable removed
--- PASS: TestConfigProvider (14.79s)
    --- PASS: TestConfigProvider/app-service-configurable (14.79s)
PASS
2023/04/03 12:04:45 [TEARDOWN]
2023/04/03 12:04:45 [exec] (sudo journalctl --since "2023-04-03 12:03:50" --no-pager | grep "edgexfoundry"|| true) > edgexfoundry.log
Wrote snap logs to /home/mengyi/Desktop/edgex671/edgex-snap-testing/test/suites/edgex-config-provider/edgexfoundry.log
2023/04/03 12:04:45 [exec] sudo snap remove --purge edgexfoundry
2023/04/03 12:04:46 [stdout] 2023-04-03T12:04:46+02:00 INFO Waiting for "snap.edgexfoundry.support-scheduler.service" to stop.
2023/04/03 12:04:52 [stdout] edgexfoundry removed
2023/04/03 12:04:52 [exec] sudo snap remove --purge edgex-config-provider-example
2023/04/03 12:04:53 [stdout] edgex-config-provider-example removed
ok      edgex-snap-testing/test/suites/edgex-config-provider    62.902s
```

**Pull request merged**